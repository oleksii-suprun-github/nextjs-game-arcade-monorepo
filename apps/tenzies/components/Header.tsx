import { PropsWithChildren } from 'react';

function Header({ children }: PropsWithChildren) {
  return (
    <header className="text-center" id="header">
      {children}
    </header>
  );
}
export default Header;
