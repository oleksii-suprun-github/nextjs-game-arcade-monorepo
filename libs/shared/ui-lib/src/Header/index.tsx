const Header = ({ children }: { children: React.ReactNode }) => (
  <div
    data-testid="header"
    className="navbar border-black/10 bg-gray-900 px-5 py-5 md:items-center md:px-10 lg:h-[80px] lg:py-0"
  >
    {children}
  </div>
);
export default Header;
