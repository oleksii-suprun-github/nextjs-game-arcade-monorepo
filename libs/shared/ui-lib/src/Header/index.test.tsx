import { describe, it, expect, vi, Mock } from 'vitest';
import { render, screen } from '@testing-library/react';
import { usePathname } from 'next/navigation';
import { createTranslator, useTranslations } from 'next-intl';
import Header from '.';

vi.mock('next/navigation', () => ({
  usePathname: vi.fn(),
}));

describe('Header', () => {
  beforeAll(async () => {
    const translate = createTranslator({
      locale: 'en',
      namespace: 'Header',
      messages: (await import('@/messages/en.json')).default,
    });

    (useTranslations as Mock).mockImplementation(() => translate);
  });

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('renders navigation links correctly', () => {
    (usePathname as Mock).mockReturnValue('/journal');
    const { container } = render(
      <Header>
        <a href="/journal">Journal</a>
        <a href="/statistics">Statistics</a>
      </Header>,
    );
    expect(container).toMatchSnapshot();

    expect(screen.getByText('Journal')).toBeInTheDocument();
    expect(screen.getByText('Statistics')).toBeInTheDocument();
  });

  it('applies active class to the current path link', () => {
    (usePathname as Mock).mockReturnValue('/journal');
    render(
      <Header>
        <a href="/journal" className="font-bold">
          Journal
        </a>
        <a href="/statistics">Statistics</a>
      </Header>,
    );

    expect(screen.getByRole('link', { name: 'Journal' })).toHaveClass('font-bold');
    expect(screen.getByRole('link', { name: 'Statistics' })).not.toHaveClass('font-bold');
  });
});
