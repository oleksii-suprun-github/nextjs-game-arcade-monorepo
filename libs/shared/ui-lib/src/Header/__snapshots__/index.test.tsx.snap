// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > renders navigation links correctly 1`] = `
<div>
  <div
    class="navbar border-black/10 bg-gray-900 px-5 py-5 md:items-center md:px-10 lg:h-[80px] lg:py-0"
    data-testid="header"
  >
    <a
      href="/journal"
    >
      Journal
    </a>
    <a
      href="/statistics"
    >
      Statistics
    </a>
  </div>
</div>
`;
